earthpy/tests/test_clip.py

```diff
@@ -19,33 +19,47 @@ def make_locs_gdf():
     return gdf
 
 
-def make_poly_in_gdf():
+def make_single_rect_poly_gdf():
     """ Bounding box polygon. """
     poly_inters = Polygon([(0, 0), (0, 10), (10, 10), (10, 0), (0, 0)])
     gdf = gpd.GeoDataFrame(
         [1], geometry=[poly_inters], crs={"init": "epsg:4326"}
     )
+    gdf["attr2"] = ["site-boundary"]
+    return gdf
+
+
+def make_smaller_clip_rect_poly_gdf():
+    """ Bounding box polygon. """
+    poly_inters = Polygon([(-5, -5), (-5, 15), (15, 15), (15, -5), (-5, -5)])
+    gdf = gpd.GeoDataFrame(
+        [1], geometry=[poly_inters], crs={"init": "epsg:4326"}
+    )
+    gdf["attr2"] = ["study area"]
     return gdf
 
 
 def make_locs_buff():
     """ Buffer points to create multi poly. """
     buffered_locations = make_locs_gdf()
     buffered_locations["geometry"] = buffered_locations.buffer(4)
+    buffered_locations["type"] = "plot"
     return buffered_locations
 
 
 def make_donut_geom():
     """ Make a donut geometry. """
     donut = gpd.overlay(
-        make_locs_buff(), make_poly_in_gdf(), how="symmetric_difference"
+        make_locs_buff(),
```

Comment:
ok another question. @mbjoseph because i use these functions, it's actually harder to test the code because i have functions wrapped in functions wrapped in fixtures. just curious what your thoughts are on this? it's kind of tedious to maintain and troubleshoot.

Reply:
See above for a way to eliminate these helper functions!

Reply:
done!!